options/cf_options.cc

```diff
@@ -280,9 +348,19 @@ std::unordered_map<std::string, OptionTypeInfo>
           OptionType::kCompressionType, OptionVerificationType::kNormal,
           OptionTypeFlags::kMutable,
           offsetof(struct MutableCFOptions, bottommost_compression)}},
-        {kNameComparator,
+        {"comparator",
          {offset_of(&ColumnFamilyOptions::comparator), OptionType::kComparator,
-          OptionVerificationType::kByName, OptionTypeFlags::kNone, 0}},
+          OptionVerificationType::kByName, OptionTypeFlags::kCompareLoose, 0,
+          [](const ConfigOptions& /*opts*/, const std::string& /*name*/,
+             const std::string& value, char* addr) {
```

Comment:
Could you add the comments for each Lambda expression to explain what it is used for? Otherwise it is hard for user to follow.